### PR TITLE
4902 imported accounts

### PR DESCRIPTION
**Pull request by andrewkryshtal**

### CODE REVIEW

File: src/components/icon-button/icon-button.tsx [hunk 1]
```diff
@@ -5,19 +5,20 @@ import { Text } from 'react-native';
...
+import { usePlusCircleButtonStyles } from './icon-button.styles';
 
 interface Props {
   text: string;
+  icon: IconNameEnum;
   onPress: EmptyFn;
 }
 
-export const PlusCircleButton: FC<Props> = ({ text, onPress }) => {
+export const IconButton: FC<Props> = ({ text, icon, onPress }) => {
```
Comment (unixvb) [inline src/components/icon-button/icon-button.tsx:16]: Need to be named according to the Temple UI Kit naming

File: src/components/mnemonic/mnemonic-input/mnemonic-input.tsx [hunk 2]
```diff
@@ -11,7 +11,13 @@ import { MnemonicProps } from '../mnemonic.props';
 import { MnemonicStyles } from '../mnemonic.styles';
 import { ProtectedOverlay } from '../protected-overlay/protected-overlay';
 
-export const MnemonicInput: FC<MnemonicProps> = ({ value, isError, onChangeText = emptyFn, onBlur = emptyFn }) => {
+export const MnemonicInput: FC<MnemonicProps> = ({
+  value,
+  isError,
+  onChangeText = emptyFn,
+  onBlur = emptyFn,
+  placeholder
```
Comment (unixvb) [inline src/components/mnemonic/mnemonic-input/mnemonic-input.tsx:19]: order

File: src/screens/manage-accounts/manage-imported-accounts/manage-imported-accounts.tsx [hunk 3]
```diff
@@ -1,11 +1,73 @@
...
-  <>
-    <InfoText />
-    <DataPlaceholder text="Imported Accounts will be available soon" />
-  </>
-);
+export const ManageImportedAccounts = () => {
+  const [managedAccount, setManagedAccount] = useState(emptyWalletAccount);
+  const { navigate } = useNavigation();
+  const accounts = useImportedAccountListSelector();
+  const { setSearchValue, filteredAccountList } = useFilteredAccountList(accounts);
```
Comment (unixvb) [inline src/screens/manage-accounts/manage-imported-accounts/manage-imported-accounts.tsx:25]: reuse this search hook on `ManageHdAccounts` screen

File: src/screens/manage-accounts/manage-imported-accounts/manage-imported-accounts.tsx [hunk 4]
```diff
@@ -1,11 +1,73 @@
...
+  };
+
+  const handleRevealButtonPress = (account: WalletAccountInterface) => {
+    setManagedAccount(account);
+    setTimeout(() => revealSelectBottomSheetController.open());
+  };
+
+  return (
+    <>
+      <SearchInput onChangeText={debouncedSetSearch} placeholder="Search accounts" />
```
Comment (unixvb) [inline src/screens/manage-accounts/manage-imported-accounts/manage-imported-accounts.tsx:48]: props order

File: src/screens/manage-accounts/manage-hd-accounts/manage-hd-accounts.tsx [hunk 5]
```diff
@@ -21,7 +22,7 @@ export const ManageHdAccounts = () => {
   const styles = useManageHdAccountsStyles();
   const revealSelectBottomSheetController = useBottomSheetController();
 
-  const hdAccounts = useHdAccountsListSelector();
+  const accounts = useHdAccountListSelector();
```
Comment (unixvb) [inline src/screens/manage-accounts/manage-hd-accounts/manage-hd-accounts.tsx:25]: rename to `hdAccounts`

File: src/modals/reveal-seed-phrase-modal/reveal-seed-phrase-form-content/reveal-seed-phrase-form-content.tsx [hunk 6]
```diff
@@ -18,15 +18,15 @@ export const RevealSeedPhraseFormContent: FC<FormikProps<RevealSeedPhraseModalFo
   values,
   setFieldValue
 }) => {
-  const hdAccounts = useHdAccountsListSelector();
+  const accounts = useAccountsListSelector();
```
Comment (unixvb) [inline src/modals/reveal-seed-phrase-modal/reveal-seed-phrase-form-content/reveal-seed-phrase-form-content.tsx:21]: `hdAccounts`

File: src/modals/reveal-private-key-modal/reveal-private-key-modal.tsx [hunk 7]
```diff
@@ -21,7 +21,7 @@ import { RevealPrivateKeyView } from './reveal-private-key-view/reveal-private-k
 export const RevealPrivateKeyModal = () => {
   const account = useRoute<RouteProp<ModalsParamList, ModalsEnum.RevealPrivateKey>>().params.account;
 
-  const hdAccounts = useHdAccountsListSelector();
+  const accounts = useAccountsListSelector();
```
Comment (unixvb) [inline src/modals/reveal-private-key-modal/reveal-private-key-modal.tsx:24]: `hdAccounts`

File: src/hooks/use-filtered-account-list.hook.ts [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+
+  useEffect(() => {
+    const result: WalletAccountInterface[] = [];
+
+    if (isString(searchValue)) {
+      const lowerCaseSearchValue = searchValue.toLowerCase();
+
+      for (const account of accountList) {
+        const { name } = account;
+        if (name.toLowerCase().includes(lowerCaseSearchValue)) {
```
Comment (unixvb) [inline src/hooks/use-filtered-account-list.hook.ts:21]: add search by `pk` & `pkh`

File: src/store/wallet/wallet-actions.ts [hunk 9]
```diff
@@ -9,7 +9,7 @@ import { TokenMetadataInterface } from '../../token/interfaces/token-metadata.in
 import { createActions } from '../create-actions';
 
 export const setSelectedAccountAction = createAction<string | undefined>('wallet/SET_SELECTED_ACCOUNT');
-export const addHdAccountAction = createAction<AccountInterface>('wallet/ADD-HD-ACCOUNT');
+export const addAccountAction = createAction<AccountInterface>('wallet/ADD-HD-ACCOUNT');
```
Comment (unixvb) [inline src/store/wallet/wallet-actions.ts:12]: this is relates to HD accounts only

File: src/store/wallet/wallet-selectors.ts [hunk 10]
```diff
@@ -12,29 +13,51 @@ import { emptyToken, TokenInterface } from '../../token/interfaces/token.interfa
...
+  useSelector<WalletRootState, WalletAccountInterface[]>(({ wallet }) => {
+    const accounts = [];
+    for (const account of wallet.accounts) {
+      if (account.type === AccountTypeEnum.IMPORTED_ACCOUNT) {
+        accounts.push(walletAccountStateToWalletAccount(account));
+      }
+    }
+
+    return accounts;
+  });
```
Comment (unixvb) [inline src/store/wallet/wallet-selectors.ts:43]: I don't think the accounts list is going to be huge.  
Just use `filter` array method instead of `for of`

File: src/components/search-input/search-input.styles.ts [hunk 11]
```diff
@@ -4,9 +4,7 @@ import { hexa } from '../../utils/style.util';
 
 export const useSearchInputStyles = createUseStyles(({ colors, typography }) => ({
   container: {
-    position: 'relative',
-    marginVertical: formatSize(8),
-    marginHorizontal: formatSize(16)
```
Comment (unixvb) [inline src/components/search-input/search-input.styles.ts:9]: remove custom paddings & move `SearchInput ` out of `ScreenContiner`

File: src/form/form-mnemonic-input.tsx [hunk 12]
```diff
@@ -7,9 +7,10 @@ import { ErrorMessage } from './error-message/error-message';
 
 interface Props {
   name: string;
+  placeholder?: string;
 }
 
-export const FormMnemonicInput: FC<Props> = ({ name }) => {
+export const FormMnemonicInput: FC<Props> = ({ name, placeholder = 'e.g. cat, dog, coffee, ocean...' }) => {
```
Comment (unixvb) [inline src/form/form-mnemonic-input.tsx:13]: move this default value up into `MnemonicInput`

File: src/modals/reveal-seed-phrase-modal/reveal-seed-phrase-form-content/reveal-seed-phrase-form-content.tsx (hunk 6 above)
Comment (unixvb) [inline src/modals/reveal-seed-phrase-modal/reveal-seed-phrase-form-content/reveal-seed-phrase-form-content.tsx:21]: here should be HD account only (as they are generated from this seed phrase)

File: src/interfaces/account.interface.ts [hunk 13]
```diff
@@ -1,17 +1,25 @@
+export enum AccountTypeEnum {
```
Comment (unixvb) [inline src/interfaces/account.interface.ts:1]: move to `enum` folder

File: src/interfaces/account.interface.ts [hunk 14]
```diff
@@ -1,17 +1,25 @@
...
+  HD_ACCOUNT = 'HD_ACCOUNT',
+  IMPORTED_ACCOUNT = 'IMPORTED_ACCOUNT'
```
Comment (unixvb) [inline src/interfaces/account.interface.ts:3]: no need in `ACCOUNT` word as it's `ACCOUNTTypeEnum`already

File: src/modals/import-account-modal/import-account-modal.tsx [hunk 15]
```diff
@@ -0,0 +1,39 @@
...
+
+  return (
+    <>
+      <ModalStatusBar />
+      {importAccountStep === 1 && (
+        <ImportAccountType
+          setImportType={setImportType}
+          importAccountStep={importAccountStep}
+          setImportAccountStep={setImportAccountStep}
+        />
```
Comment (unixvb) [inline src/modals/import-account-modal/import-account-modal.tsx:26]: instead of passing 3 props, just use `onSubmit` props that will return `type` as payload. (and move all that logic form `ImportAccountType` here)

File: src/modals/import-account-modal/import-account-type/import-account-type.tsx [hunk 16]
```diff
@@ -0,0 +1,61 @@
...
+  setImportAccountStep: Dispatch<SetStateAction<number>>;
+  setImportType: Dispatch<SetStateAction<ImportAccountTypeEnum>>;
+}
+
+export const ImportAccountType: FC<Props> = ({ importAccountStep, setImportAccountStep, setImportType }) => {
+  const { goBack } = useNavigation();
+  const typeRadioButtons: RadioButton<ImportAccountTypeEnum>[] = [
+    { value: ImportAccountTypeEnum.PRIVATE_KEY, label: 'Private key' },
+    { value: ImportAccountTypeEnum.SEED_PHRASE, label: 'Seed phrase' }
+  ];
```
Comment (unixvb) [inline src/modals/import-account-modal/import-account-type/import-account-type.tsx:30]: move this variable out of component (it have no deps) to not let it reinitialise on each rerender

File: src/modals/import-account-modal/import-account-modal.form.tsx [hunk 17]
```diff
@@ -0,0 +1,42 @@
...
+export const importAccountSeedFormValidationSchema: SchemaOf<ImportAccountSeedValues> = object().shape({
+  seedPhrase: seedPhraseValidation,
+  password: string(),
+  derivation: mixed<ImportAccountDerivationEnum>().oneOf(Object.values(ImportAccountDerivationEnum)),
+  derivationPath: string().required()
+});
+
+export const importAccountPrivateKeyFormValidationSchema = object().shape({
+  privateKey: string().required(makeRequiredErrorMessage('Private key'))
+});
```
Comment (unixvb) [inline src/modals/import-account-modal/import-account-modal.form.tsx:42]: divide this file into part (for each component seperatly)

File: src/modals/import-account-modal/import-account-seed-form/import-account-seed-form.tsx [hunk 18]
```diff
@@ -0,0 +1,96 @@
...
+import { FormTextInput } from '../../../form/form-text-input';
+import { ImportAccountDerivationEnum, ImportAccountSeedValues } from '../../../interfaces/import-account-type';
+import { useShelter } from '../../../shelter/use-shelter.hook';
+import { useAccountsListSelector } from '../../../store/wallet/wallet-selectors';
+import { formatSize } from '../../../styles/format-size';
+import {
+  importAccountSeedFormInitialValues,
+  importAccountSeedFormValidationSchema
+} from '../import-account-modal.form';
+import { useImportAccountModalStyles } from '../import-account-modal.styles';
```
Comment (unixvb) [inline src/modals/import-account-modal/import-account-seed-form/import-account-seed-form.tsx:25]: rename variable & file names (mismatch with component name)

File: src/modals/import-account-modal/import-account-private-key-form/import-account-private-key-form.tsx [hunk 19]
```diff
@@ -0,0 +1,61 @@
...
+import { useAccountsListSelector } from '../../../store/wallet/wallet-selectors';
+import { formatSize } from '../../../styles/format-size';
+import {
+  importAccountPrivateKeyFormInitialValues,
+  importAccountPrivateKeyFormValidationSchema
+} from '../import-account-modal.form';
+
+interface Props {
+  importAccountStep: number;
+  setImportAccountStep: Dispatch<SetStateAction<number>>;
```
Comment (unixvb) [inline src/modals/import-account-modal/import-account-private-key-form/import-account-private-key-form.tsx:24]: do not pass this prop into child component (return submit/press handlers). 
take a look at `AddTokenModal` as example

File: src/modals/import-account-modal/import-account-seed-form/import-account-seed-form.tsx [hunk 20]
```diff
@@ -0,0 +1,96 @@
...
+
+interface Props {
+  importAccountStep: number;
+  setImportAccountStep: Dispatch<SetStateAction<number>>;
```
Comment (unixvb) [inline src/modals/import-account-modal/import-account-seed-form/import-account-seed-form.tsx:29]: do not pass this prop into child component (return submit/press handlers). 
take a look at `AddTokenModal` as example

File: src/modals/import-account-modal/import-account-seed-form/import-account-seed-form.tsx [hunk 21]
```diff
@@ -0,0 +1,96 @@
...
+}
+
+export const ImportAccountSeedForm: FC<Props> = ({ setImportAccountStep, importAccountStep }) => {
+  const styles = useImportAccountModalStyles();
+  const { createImportedAccountWithSeed } = useShelter();
+  const accountsLength = useAccountsListSelector().length + 1;
+  const derivationRadioButtons: RadioButton<ImportAccountDerivationEnum>[] = [
+    { value: ImportAccountDerivationEnum.DEFAULT, label: 'Default account (the first one)' },
+    { value: ImportAccountDerivationEnum.CUSTOM_PATH, label: 'Custom derivation path' }
+  ];
```
Comment (unixvb) [inline src/modals/import-account-modal/import-account-seed-form/import-account-seed-form.tsx:39]: the same (move out of component)

File: src/interfaces/import-account-type.ts [hunk 22]
```diff
@@ -0,0 +1,24 @@
...
+}
+
+export type ImportAccountTypeValues = {
+  type: ImportAccountTypeEnum;
+};
+
+export type ImportAccountSeedValues = {
+  seedPhrase: string;
+  password?: string;
+  derivation?: string;
```
Comment (unixvb) [inline src/interfaces/import-account-type.ts:18]: why `derivation` is optional and why not `ImportAccountDerivationEnum` type?
also maybe rename to `derivationType`?

File: src/modals/import-account-modal/import-account-seed-form/import-account-seed-form.tsx [hunk 23]
```diff
@@ -0,0 +1,96 @@
...
+
+  const onSubmit = (values: ImportAccountSeedValues) =>
+    createImportedAccountWithSeed({
+      name: `Account ${accountsLength}`,
+      seedPhrase: values.seedPhrase,
+      password: values.password,
+      derivationPath: values.derivationPath
```
Comment (unixvb) [inline src/modals/import-account-modal/import-account-seed-form/import-account-seed-form.tsx:46]: check here for `derivationType` value as well. (User could change `derivationPath` and switch back to `ImportAccountDerivationEnum.DEFAULT`)

File: src/modals/import-account-modal/import-account-private-key-form/import-account-private-key-form.tsx [hunk 24]
```diff
@@ -0,0 +1,61 @@
...
+}
+
+export const ImportAccountPrivateKeyForm: FC<Props> = ({ importAccountStep, setImportAccountStep }) => {
+  const { createImportedAccountWithPrivateKey } = useShelter();
+  const accountLength = useAccountsListSelector().length + 1;
```
Comment (unixvb) [inline src/modals/import-account-modal/import-account-private-key-form/import-account-private-key-form.tsx:29]: you are adding `+ 1`, it's not `accountLength` any more (`accountIndex` for eg.)

File: src/modals/import-account-modal/import-account-seed-form/import-account-seed-form.tsx [hunk 25]
```diff
@@ -0,0 +1,96 @@
...
+
+interface Props {
+  importAccountStep: number;
+  setImportAccountStep: Dispatch<SetStateAction<number>>;
+}
+
+export const ImportAccountSeedForm: FC<Props> = ({ setImportAccountStep, importAccountStep }) => {
+  const styles = useImportAccountModalStyles();
+  const { createImportedAccountWithSeed } = useShelter();
+  const accountsLength = useAccountsListSelector().length + 1;
```
Comment (unixvb) [inline src/modals/import-account-modal/import-account-seed-form/import-account-seed-form.tsx:35]: you are adding `+ 1`, it's not `accountLength` any more (`accountIndex` for eg.)

File: src/shelter/shelter.ts [hunk 26]
```diff
@@ -89,17 +116,52 @@ export class Shelter {
...
+        showErrorToast({
+          title: 'Failed to import account.',
+          description: 'This may happen because provided Key is invalid.'
+        });
+
+        return of(undefined);
+      })
+    );
+  };
+
```
Comment (unixvb) [inline src/shelter/shelter.ts:152]: a lot of duplication.
create 1 method that will import Imported account (for both cases) and just prepare correct data for seed case

File: src/modals/import-account-modal/import-account-seed-form/import-account-seed-form.tsx (hunk 23 above)
Comment (andrewkryshtal) [reply src/modals/import-account-modal/import-account-seed-form/import-account-seed-form.tsx:46]: and there will be no value in `derivationPath`, will be used default value

File: src/components/search-input/search-input.tsx [hunk 27]
```diff
@@ -9,7 +9,9 @@ import { Icon } from '../icon/icon';
 import { IconNameEnum } from '../icon/icon-name.enum';
 import { useSearchInputStyles } from './search-input.styles';
 
-type Props = Pick<TextInputProps, 'value' | 'placeholder' | 'onChangeText'>;
+interface Props extends Pick<TextInputProps, 'value' | 'placeholder' | 'onChangeText'> {
+  containerStyle?: ViewStyle;
+}
```
Comment (unixvb) [inline src/components/search-input/search-input.tsx:14]: no need in this anymore

File: src/components/search-input/search-input.tsx [hunk 28]
```diff
@@ -18,7 +20,7 @@ export const SearchInput: FC<Props> = ({ value, placeholder, onChangeText = empt
   const debouncedOnChangeText = debounce(onChangeText);
 
   return (
-    <View style={styles.container}>
+    <View style={[styles.container]}>
```
Comment (unixvb) [inline src/components/search-input/search-input.tsx:23]: the same

File: src/components/segmented-control/segmented-control.styles.ts [hunk 29]
```diff
@@ -13,7 +13,8 @@ export const useSegmentedControlStyles = createUseStyles(({ colors }) => ({
     height: formatSize(28),
     borderRadius: formatSize(8),
     backgroundColor: colors.lines,
-    padding: formatSize(2)
+    padding: formatSize(2),
+    marginHorizontal: formatSize(16)
```
Comment (unixvb) [inline src/components/segmented-control/segmented-control.styles.ts:17]: would not this break other `SegmentedControls`?

File: src/modals/import-account-modal/import-account-modal.tsx [hunk 30]
```diff
@@ -0,0 +1,37 @@
...
+import { ModalStatusBar } from '../../components/modal-status-bar/modal-status-bar';
+import { ImportAccountTypeEnum, ImportAccountTypeValues } from '../../interfaces/import-account-type';
+import { ImportAccountPrivateKey } from './import-account-private-key/import-account-private-key';
+import { ImportAccountSeed } from './import-account-seed/import-account-seed';
+import { ImportAccountType } from './import-account-type/import-account-type';
+
+export const ImportAccountModal = () => {
+  useNavigationSetOptions({
+    headerLeft: () => <HeaderProgress current={importAccountStep} total={2} />
+  });
```
Comment (unixvb) [inline src/modals/import-account-modal/import-account-modal.tsx:14]: do counter changes without `importAccountStep` in deps?

File: src/modals/import-account-modal/import-account-type/import-account-type.tsx [hunk 31]
```diff
@@ -0,0 +1,56 @@
...
+        <ScreenContainer isFullScreenMode={true}>
+          <View>
+            <Divider size={formatSize(12)} />
+            <Label label="Type of import" description="Select how would you like to import account." />
+            <FormRadioButtonsGroup
+              name="type"
+              buttons={[
+                { value: ImportAccountTypeEnum.PRIVATE_KEY, label: 'Private key' },
+                { value: ImportAccountTypeEnum.SEED_PHRASE, label: 'Seed phrase' }
+              ]}
```
Comment (unixvb) [inline src/modals/import-account-modal/import-account-type/import-account-type.tsx:41]: this will create new buttons and cause rerender of `FormRadioButtonsGroup ` on each render of `ImportAccountType` component.  
move this into a variable and move out of component

File: src/modals/import-account-modal/import-account-type/import-account-type.tsx [hunk 32]
```diff
@@ -0,0 +1,56 @@
...
+import { FormRadioButtonsGroup } from '../../../form/form-radio-buttons-group';
+import { ImportAccountTypeEnum, ImportAccountTypeValues } from '../../../interfaces/import-account-type';
+import { useNavigation } from '../../../navigator/hooks/use-navigation.hook';
+import { formatSize } from '../../../styles/format-size';
+import { importAccountTypeInitialValues, importAccountTypeValidationSchema } from './import-account-type.form';
+
+interface Props {
+  onSubmit: EventFn<ImportAccountTypeValues>;
+}
+
```
Comment (unixvb) [inline src/modals/import-account-modal/import-account-type/import-account-type.tsx:22]: here for ex.
https://github.com/madfish-solutions/templewallet-mobile/pull/172/files#r684012255

File: src/modals/import-account-modal/import-account-seed/import-account-seed.tsx [hunk 33]
```diff
@@ -0,0 +1,102 @@
...
+              label="Derivation"
+              isOptional
+              description="By default derivation isn't used. Click on 'Custom derivation path' to add it."
+            />
+            <FormRadioButtonsGroup
+              name="derivationType"
+              buttons={[
+                { value: ImportAccountDerivationEnum.DEFAULT, label: 'Default account (the first one)' },
+                { value: ImportAccountDerivationEnum.CUSTOM_PATH, label: 'Custom derivation path' }
+              ]}
```
Comment (unixvb) [inline src/modals/import-account-modal/import-account-seed/import-account-seed.tsx:81]: the same
https://github.com/madfish-solutions/templewallet-mobile/pull/172/files#r684012255

File: src/modals/import-account-modal/import-account-seed/import-account-seed.tsx [hunk 34]
```diff
@@ -0,0 +1,102 @@
...
+              label="Password"
+              isOptional
+              description={'Used for additional mnemonic derivation.\nThat is NOT a wallet password.'}
+            />
+            <FormPasswordInput name="password" />
+            <Divider size={formatSize(12)} />
+            <Label
+              label="Derivation"
+              isOptional
+              description="By default derivation isn't used. Click on 'Custom derivation path' to add it."
```
Comment (unixvb) [inline src/modals/import-account-modal/import-account-seed/import-account-seed.tsx:74]: use same codestyle for `description` prop

File: src/shelter/shelter.ts [hunk 35]
```diff
@@ -89,17 +92,43 @@ export class Shelter {
...
+            publicKey,
+            publicKeyHash
+          })
+        )
+      ),
+      catchError(() => {
+        showErrorToast({
+          title: 'Failed to import account.',
+          description: 'This may happen because provided Key is invalid.'
+        });
```
Comment (unixvb) [inline src/shelter/shelter.ts:113]: show this error toast inside `useShelterHook` subscription

File: src/shelter/use-shelter.hook.ts [hunk 36]
```diff
@@ -38,6 +39,7 @@ export const useShelter = () => {
           )
         )
         .subscribe(([publicData, isPasswordSaved]) => {
+          console.log({ publicData });
```
Comment (unixvb) [inline src/shelter/use-shelter.hook.ts:42]: remove

File: src/screens/manage-accounts/manage-hd-accounts/manage-hd-accounts.tsx [hunk 37]
```diff
@@ -21,7 +26,11 @@ export const ManageHdAccounts = () => {
   const styles = useManageHdAccountsStyles();
   const revealSelectBottomSheetController = useBottomSheetController();
 
-  const hdAccounts = useHdAccountsListSelector();
+  const hdAccounts = useHdAccountListSelector();
+  const { setSearchValue, filteredAccountList } = useFilteredAccountList(hdAccounts);
+
+  const debouncedSetSearch = debounce(setSearchValue);
```
Comment (unixvb) [inline src/screens/manage-accounts/manage-hd-accounts/manage-hd-accounts.tsx:32]: the same behaviour in 2 places.  
move `debouncedSetSearch` inside `useFilteredAccountList`

File: src/components/segmented-control/segmented-control.styles.ts [hunk 38]
```diff
@@ -12,8 +12,7 @@ export const useSegmentedControlStyles = createUseStyles(({ colors }) => ({
     position: 'relative',
     height: formatSize(28),
     borderRadius: formatSize(8),
-    backgroundColor: colors.lines,
-    padding: formatSize(2)
```
Comment (unixvb) [inline src/components/segmented-control/segmented-control.styles.ts:16]: revert this change

File: src/interfaces/import-account-type.ts [hunk 39]
```diff
@@ -0,0 +1,24 @@
...
+export type ImportAccountSeedValues = {
+  seedPhrase: string;
+  password?: string;
+  derivationType: ImportAccountDerivationEnum;
+  derivationPath: string;
+};
+
+export type ImportAccountPrivateKeyValues = {
+  privateKey: '';
+};
```
Comment (unixvb) [inline src/interfaces/import-account-type.ts:24]: split by files pls

**Review by unixvb — CHANGES_REQUESTED** [review]

File: src/modals/import-account-modal/import-account-type/import-account-type.tsx [hunk 40]
```diff
@@ -10,14 +10,16 @@ import { InsetSubstitute } from '../../../components/inset-substitute/inset-subs
...
 import { FormRadioButtonsGroup } from '../../../form/form-radio-buttons-group';
-import { ImportAccountTypeEnum, ImportAccountTypeValues } from '../../../interfaces/import-account-type';
 import { useNavigation } from '../../../navigator/hooks/use-navigation.hook';
 import { formatSize } from '../../../styles/format-size';
 import { importAccountTypeInitialValues, importAccountTypeValidationSchema } from './import-account-type.form';
 
 interface Props {
-  onSubmit: EventFn<ImportAccountTypeValues>;
+  onSubmit: EventFn<{
+    type: ImportAccountTypeEnum;
```
Comment (unixvb) [inline src/modals/import-account-modal/import-account-type/import-account-type.tsx:21]: maybe just simplify this to `EventFn<ImportAccountTypeEnum>` in 2 places?

**Review by keshan3262 — APPROVED** [review]

**Review by unixvb — APPROVED** [review]
